### PR TITLE
Add example code for thirpartyemailpassword

### PR DESCRIPTION
**Pull request by nkshah2**
## Summary of change

Adds an example for Social auth which uses Google, Github and Apple sign in.

## Test Plan

Tested this server code against:
- React Native (Android and iOS)
- ReactJS

## Documentation changes
- [x] Code commenting

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [ ] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`). If creating a sub recipe, make sure to use `bind(this)` when calling the original implementation from your functions

## Remaining TODOs for this PR
- [x] Add code comments

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/node_modules
+node_modules
```
Comment (rishabhpoddar) [inline .gitignore:1]: Undo this. We want a .gitignore in the example app for it self.

File: examples/with-thirdpartyemailpassword-express/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+# SuperTokens ThirdPartyEmailPassword Demo app
+
+This server code can be used to implement social auth using:
+
+-   Google
+-   Github
+-   Apple
+
+For React Native apps (Android and iOS) and Websites
```
Comment (rishabhpoddar) [inline examples/with-thirdpartyemailpassword-express/README.md:11]: Don't need to mention this line

File: examples/with-thirdpartyemailpassword-express/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,583 @@
+{
```
Comment (rishabhpoddar) [inline examples/with-thirdpartyemailpassword-express/package-lock.json:1]: This file should not be on git

File: examples/with-thirdpartyemailpassword-express/package.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    "scripts": {
+        "start": "node index.js"
+    },
+    "dependencies": {
+        "cors": "^2.8.5",
+        "dotenv": "^8.6.0",
+        "express": "^4.17.1",
+        "helmet": "^4.4.0",
+        "morgan": "^1.10.0",
+        "supertokens-node": "git+https://github.com/supertokens/supertokens-node.git#8.1"
```
Comment (rishabhpoddar) [inline examples/with-thirdpartyemailpassword-express/package.json:15]: Changed to ^8.1.0 after its released

File: examples/with-thirdpartyemailpassword-express/index.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                }),
+                ThirdPartyEmailPassword.Github({
+                    clientSecret: "00e841f10f288363cd3786b1b1f538f05cfdbda2",
+                    clientId: "8a9152860ce869b64c44",
+                }),
+                /*
+                    For Apple signin, iOS apps always use the bundle identifier as the client ID when communicating with Apple. Android, Web and other platforms
+                    need to configure a Service ID on the Apple developer dashboard and use that as client ID.
+
+                    In the example below 4398792-io.supertokens.example.service is the client ID for Web. Android etc and thus we mark it as default. For iOS
```
Comment (rishabhpoddar) [inline examples/with-thirdpartyemailpassword-express/index.js:65]: Should be a comma instead of a dot in "Web. Android"

Comment (rishabhpoddar) [reply examples/with-thirdpartyemailpassword-express/index.js:65]: Remove "and thus we mark it as default"

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (nkshah2) [reply .gitignore]: Done

File: examples/with-thirdpartyemailpassword-express/README.md (hunk 2 above)
Comment (nkshah2) [reply examples/with-thirdpartyemailpassword-express/README.md]: Removed

File: examples/with-thirdpartyemailpassword-express/package-lock.json (hunk 3 above)
Comment (nkshah2) [reply examples/with-thirdpartyemailpassword-express/package-lock.json]: Done

File: examples/with-thirdpartyemailpassword-express/index.js (hunk 5 above)
Comment (nkshah2) [reply examples/with-thirdpartyemailpassword-express/index.js]: Done

**Pull request merged**